### PR TITLE
Show kubernetes version in cluster overview

### PR DESCRIPTION
**Pull request by marians**
Closes https://github.com/giantswarm/giantswarm/issues/13478

This PR adds the k8s version number to the cluster overview.

## Before

image

## After

image

### CODE REVIEW

File: src/components/Home/ClusterDashboardItem.js [hunk 1]
```diff
@@ -251,6 +254,22 @@ function ClusterDashboardItem({
               </span>
             </RefreshableLabel>
             <Dot style={{ paddingLeft: 0 }} />
+            <RefreshableLabel value={cluster.kubernetes_version}>
```
Comment (ubergesundheit) [inline src/components/Home/ClusterDashboardItem.js:257]: there is no `kubernetes_version` field in clusters

File: src/components/Home/ClusterDashboardItem.js [hunk 2]
```diff
@@ -251,6 +254,22 @@ function ClusterDashboardItem({
...
+                <i className='fa fa-kubernetes' title='Kubernetes version' />{' '}
+                <span>
+                  {(() => {
+                    const kubernetes = release.components.find(
+                      (component) => component.name === 'kubernetes'
+                    );
+                    if (kubernetes) return kubernetes.version;
+
+                    return null;
+                  })()}
```
Comment (ubergesundheit) [inline src/components/Home/ClusterDashboardItem.js:268]: The `release` contains the field `kubernetesVersion` which should be used instead of this.

File: src/components/Home/ClusterDashboardItem.js (hunk 1 above)
Comment (marians) [reply src/components/Home/ClusterDashboardItem.js:257]: Right, changing

File: src/components/Home/ClusterDashboardItem.js (hunk 2 above)
Comment (marians) [reply src/components/Home/ClusterDashboardItem.js:268]: Oh. Then I copied a bad example :)

Comment (axbarsan) [inline src/components/Home/ClusterDashboardItem.js:268]: Can you please extract this into a separate function, this is a pattern that we're trying to get rid of?

**Review by ubergesundheit — CHANGES_REQUESTED** [review]
Nice addition. Implementation is a little rough around the edges

File: src/components/Home/ClusterDashboardItem.js (hunk 2 above)
Comment (marians) [reply src/components/Home/ClusterDashboardItem.js:268]: Gerald already mentioned that it can be replaced simply with `release.kubernetesVersion`.

**Comment by marians** [comment]
There is a lot going on in the tests which I don't understand. Maybe somebody could help me here?

File: src/components/Home/ClusterDashboardItem.js [hunk 3]
```diff
@@ -251,6 +254,13 @@ function ClusterDashboardItem({
               </span>
             </RefreshableLabel>
             <Dot style={{ paddingLeft: 0 }} />
+            <RefreshableLabel value={release.kubernetesVersion}>
```
Comment (axbarsan) [inline src/components/Home/ClusterDashboardItem.js:257]: The problem is possibly because if the release version is not found, then the value of `release` is `null`, which doesn't have a `kubernetesVersion` property.

This should be enough here.

```suggestion
            <RefreshableLabel value={release?.kubernetesVersion}>
```

**Comment by marians** [comment]
There is also this problem:

image

This remains with my branch, while it's not happening in current master.

File: src/components/Home/ClusterDashboardItem.js [hunk 4]
```diff
@@ -251,6 +254,13 @@ function ClusterDashboardItem({
...
+              <span>
+                <i className='fa fa-kubernetes' title='Kubernetes version' />{' '}
+                {release.kubernetesVersion}
```
Comment (axbarsan) [inline src/components/Home/ClusterDashboardItem.js:260]: However, inside the `span` element, you might want to add a placeholder

```suggestion
                <i className='fa fa-kubernetes' title='Kubernetes version' />{' '}
                {release?.kubernetesVersion ?? 'N/A'}
```

File: src/components/Home/ClusterDashboardItem.js [hunk 5]
```diff
@@ -143,10 +144,12 @@ function ClusterDashboardItem({
   dispatch,
   nodePoolsLoadError,
   clusterId,
+  releases,
 }) {
   // If the cluster has been deleted using gsctl, Happa doesn't know yet.
   if (!cluster) return null;
 
+  const release = releases[cluster.release_version] ?? null;
```
Comment (ubergesundheit) [inline src/components/Home/ClusterDashboardItem.js:152]: Maybe the errors in the tests are related to this line. `null` does not support being accessed (Line 257 - 263)

Comment (marians) [reply src/components/Home/ClusterDashboardItem.js:152]: So what should I do?

File: src/components/Home/ClusterDashboardItem.js (hunk 3 above)
Comment (marians) [reply src/components/Home/ClusterDashboardItem.js]: Done

File: src/components/Home/ClusterDashboardItem.js (hunk 4 above)
Comment (marians) [reply src/components/Home/ClusterDashboardItem.js]: Done

**Comment by marians** [comment]
Tests are green now, but this keeps showing up as zero:

image

File: src/components/Cluster/ClusterDetail/RegionAndVersions.js [hunk 6]
```diff
@@ -66,14 +66,7 @@ function RegionAndVersions({
...
               <i className='fa fa-kubernetes' />
-              {(() => {
-                const kubernetes = release.components.find(
-                  (component) => component.name === 'kubernetes'
-                );
-                if (kubernetes) return kubernetes.version;
-
-                return null;
-              })()}
+              {release.kubernetesVersion}
```
Comment (ubergesundheit) [inline src/components/Cluster/ClusterDetail/RegionAndVersions.js:69]: Could you please undo this change? I'm preparing a PR which rewrites this component in typescript lined up and would like to avoid having merge conflicts

**Comment by marians** [comment]
Ack, thanks!

File: src/components/Home/ClusterDashboardItem.js (hunk 5 above)
Comment (ubergesundheit) [reply src/components/Home/ClusterDashboardItem.js:152]: You already solved this by adding `?` to the mentioned lines

File: src/components/Cluster/ClusterDetail/RegionAndVersions.js (hunk 6 above)
Comment (marians) [reply src/components/Cluster/ClusterDetail/RegionAndVersions.js:69]: If that makes you happy :) Done

**Review by ubergesundheit — APPROVED** [review]
:bow:
